app/src/main/java/org/literacyapp/service/synchronization/MergeSimilarStudentsJobService.java

```diff
@@ -1,30 +1,27 @@
 package org.literacyapp.service.synchronization;
 
-import android.app.Service;
 import android.app.job.JobParameters;
 import android.app.job.JobService;
-import android.content.Intent;
-import android.os.IBinder;
 import android.util.Log;
 
-import org.literacyapp.authentication.TrainingHelper;
+import org.literacyapp.authentication.MergeThread;
+import org.literacyapp.authentication.TrainingThread;
 
 public class MergeSimilarStudentsJobService extends JobService {
-    public static boolean isRunning = false;
+    private MergeThread mergeThread;
 
     @Override
     public boolean onStartJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStartJob");
-        isRunning = true;
-        TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
-        trainingHelper.findAndMergeSimilarStudents();
-        isRunning = false;
+        mergeThread = new MergeThread(getApplicationContext());
```

Comment:
I haven't done this before, but it seems like you can create an AsyncTask in the same Java class: http://toastdroid.com/2015/02/21/how-to-use-androids-job-scheduler/